core/controllers/reader.py

```diff
@@ -274,15 +274,45 @@ def get(self, exploration_id):
         exploration_rights = rights_manager.get_exploration_rights(
             exploration_id, strict=False)
         user_settings = user_services.get_user_settings(self.user_id)
+        exploration_user_data = exp_fetchers.get_exploration_user_data(
+            self.user_id, exploration_id)
 
         preferred_audio_language_code = None
         preferred_language_codes = None
+        user_has_viewed_lesson_info_modal_once = None
 
         if user_settings is not None:
             preferred_audio_language_code = (
                 user_settings.preferred_audio_language_code)
             preferred_language_codes = (
                 user_settings.preferred_language_codes)
+            user_has_viewed_lesson_info_modal_once = (
+                user_settings.user_has_viewed_lesson_info_modal_once)
+
+        furthest_reached_checkpoint_exp_version = None
+        furthest_reached_checkpoint_state_name = None
+        most_recently_reached_checkpoint_exp_version = None
+        most_recently_reached_checkpoint_state_name = None
+
+        if exploration_user_data is not None:
+            # If the most recently reached checkpoint is not present in the
```

Comment:
Yes, we are syncing learner's checkpoint progress when the exploration has been updated since he last interacted it with. I have updated the comment now to reflect the same.